src/Raven.Server/Documents/Indexes/MapReduce/MapReduceIndexingContext.cs

```diff
@@ -40,24 +43,41 @@ static MapReduceIndexingContext()
 
         public void Dispose()
         {
-            StoreNextMapResultId();
-            DocumentMapEntries?.Dispose();
-            DocumentMapEntries = null;
-            MapPhaseTree = null;
-            ReducePhaseTree = null;
-            PageModifiedInReduceTree = null;
-            ProcessedDocEtags.Clear();
-            ProcessedTombstoneEtags.Clear();
-            StoreByReduceKeyHash.Clear();
+            try
+            {
+                StoreNextMapResultId();
+            }
+            finally
+            {
+                StoreNextMapResultId();
```

Comment:
Why are you calling `StoreNextMapResultId` twice? Once in the `try` and once in the `finally` ?

Reply:
This wasn't ported correctly from v4.1. Fixed.